tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -0,0 +1,196 @@
+import time
+import json
+from enum import IntEnum
+
+from rptest.utils.rpenv import sample_license
+from rptest.services.admin import Admin, EnterpriseLicenseStatus, RolesList, RoleDescription
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST, SecurityConfig, SchemaRegistryConfig
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.cluster import cluster
+from rptest.services.redpanda_installer import RedpandaInstaller, wait_for_num_versions
+from rptest.util import expect_exception
+
+from ducktape.errors import TimeoutError as DucktapeTimeoutError
+from ducktape.utils.util import wait_until
+from ducktape.mark import parametrize, matrix
+from rptest.util import wait_until_result
+
+
+class EnterpriseFeaturesTestBase(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.admin = Admin(self.redpanda)
+        self.installer = self.redpanda._installer
+
+    def setUp(self):
+        super().setUp()
+
+
+class Features(IntEnum):
+    audit_logging = 0
+    cloud_storage = 1
+    partition_auto_balancing_continuous = 2
+    core_balancing_continuous = 3
+    gssapi = 4
+    oidc = 5
+    schema_id_validation = 6
+    rbac = 7
+    fips = 8
+
+
+SKIP_FEATURES = [
+    Features.audit_logging,  # NOTE(oren): omit due to shutdown issues
+    Features.
+    cloud_storage,  # TODO(oren): initially omitted because it's a bit complicated to initialize infra
+]
+
+
+class EnterpriseFeaturesTest(EnterpriseFeaturesTestBase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            num_brokers=3,
+            schema_registry_config=SchemaRegistryConfig(),
+            **kwargs,
+        )
+
+        self.su, self.pw, self.mech = self.redpanda.SUPERUSER_CREDENTIALS
+
+        self.security = SecurityConfig()
+        self.security.enable_sasl = True
+        self.kafka_enable_authorization = True
+        self.endpoint_authn_method = 'sasl'
+
+        self.redpanda.set_security_settings(self.security)
+
+    @cluster(num_nodes=3)
+    def test_get_enterprise(self):
+        rsp = self.admin.get_enterprise_features().json()
+
+        status = rsp.get('license_status', None)
+        assert type(status) == str, f"Ill-formed license_status {type(status)}"
+        assert EnterpriseLicenseStatus.from_str(
+            status) is not None, f"Unexpected status '{status}'"
+
+        violation = rsp.get('violation', None)
+        assert type(
+            violation) == bool, f"Ill-formed violation flag {type(violation)}"
+        assert not violation, "Config unexpectedly in violation"
```

Comment:
fix/test: will probably in FIPS CDT as FIPS is enabled in the FIPS CDT environment

Reply:
right, right, we talked about this the other day. per below, will give a run through CDT and see what happens.

Reply:
added `fips_ok_to_fail` to avoid too much mucking around. lmk if you think it's worth refactoring the tests to account for this.